test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -96,7 +97,32 @@ private void ConfigureEchoAddress(IApplicationBuilder app)
         {
             app.Run(context =>
             {
-                return context.Response.WriteAsync(context.Request.GetDisplayUrl());
+                const string SchemeDelimiter = "://";
+                
+                var request = context.Request;
+                var host = request.Host.Value;
+                var pathBase = request.PathBase.Value;
+                var path = request.Path.Value;
+                var queryString = request.QueryString.Value;
+
+                // PERF: Calculate string length to allocate correct buffer size for StringBuilder.
```

Comment:
Need to revert this. This was copied from elsewhere when I was investigating a null ref a test was throwing.